File: extracted/ru/runlayer/tests/fixtures/agent_detection/samples/sample_07/agent.py
"""Minimal OpenAI Agents SDK agent (framework detection corpus)."""

from agents import Agent, Runner, function_tool


@function_tool
def get_weather(city: str) -> str:
    """Get the current weather for a city."""
    return f"It's 72F and sunny in {city}."


@function_tool
def add(a: int, b: int) -> int:
    """Add two integers."""
    return a + b


def main() -> None:
    agent = Agent(
        name="Assistant",
        instructions="You are a helpful assistant. Use tools when needed.",
        tools=[get_weather, add],
    )
    result = Runner.run_sync(agent, "What's the weather in Paris, and what is 21 + 21?")
    print(result.final_output)


if __name__ == "__main__":
    main()

File: extracted/ru/runlayer/tests/fixtures/agent_detection/samples/sample_12/agent.ts
/** Minimal OpenAI Agents SDK (JS) agent (framework detection corpus). */

import { Agent, run, tool } from "@openai/agents";
import { z } from "zod";

const getWeather = tool({
  name: "get_weather",
  description: "Get the current weather for a city.",
  parameters: z.object({ city: z.string() }),
  execute: async ({ city }) => `It's 72F and sunny in ${city}.`,
});

const add = tool({
  name: "add",
  description: "Add two integers.",
  parameters: z.object({ a: z.number(), b: z.number() }),
  execute: async ({ a, b }) => `${a + b}`,
});

async function main(): Promise<void> {
  const agent = new Agent({
    name: "Assistant",
    instructions: "You are a helpful assistant. Use tools when needed.",
    tools: [getWeather, add],
  });
  const result = await run(
    agent,
    "What's the weather in Paris, and what is 21 + 21?",
  );
  console.log(result.finalOutput);
}

void main();
